pyqldb/driver/qldb_driver.py

```diff
@@ -309,39 +327,28 @@ def _get_session(self):
 
         logger.debug('Getting session. Current free session count: {}. Current available permit count: {}.'.format(
             self._pool.qsize(), self._pool_permits_counter.value))
+
         if self._pool_permits.acquire(timeout=self._timeout):
             self._pool_permits_counter.decrement()
+
+            session = None
             try:
-                try:
-                    cur_session = self._pool.get_nowait()
-                    logger.debug('Reusing session from pool. Session ID: {}.'.format(cur_session.session_id))
-                    return cur_session
-                except Empty:
-                    pass
-
-                logger.debug('Creating new session.')
-                return self._create_new_session()
+                if not start_new_session:
+                    try:
+                        session = self._pool.get_nowait()
+                        logger.debug('Reusing session from pool. Session ID: {}.'.format(session.session_id))
+                        return session
+                    except Empty:
+                        pass
+
+                if start_new_session or session is None:
```

Comment:
I think the 'session is None' check here would always be true? Was it meant perhaps as check to see if the returned session from the pool was none?

Reply:
It was to include the case where getting session from pool was unsuccessful. 

I have simplified the logic and we don't need to have the if condition